### PR TITLE
Add Object.entries polyfill

### PR DESCRIPTION
**Pull request by JakeChampion**

### CODE REVIEW

File: polyfills/Object/entries/polyfill.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+	global.Object.entries = function entries(object) {
+		var keys = Object.keys(object);
+
+		return keys
+			.reduce(function (entries, key) {
+				var entry = typeof key === 'string' && object.propertyIsEnumerable(key) ? [[key, object[key]]] : [];
+				return entries.concat(entry);
+			}, []);
+	};
+}(this));
```
Comment (triblondon) [inline polyfills/Object/entries/polyfill.js]: wrapper unnecessary?

Comment (triblondon) [reply polyfills/Object/entries/polyfill.js]: The polyfill service already offers a guarantee that `this` is a global

File: polyfills/Object/entries/config.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+		"op_mini": "*",
+		"samsung_mob": "*",
+		"bb": "*"
+},
+"dependencies": [
+	"Array.prototype.reduce",
+	"Object.keys"
+],
+"docs": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries"
+}
```
Comment (triblondon) [inline polyfills/Object/entries/config.json:26]: Indentation issues

**Review by triblondon — APPROVED** [review]
Subject to changes noted in review

File: polyfills/Object/entries/config.json (hunk 2 above)
Comment (JakeChampion) [reply polyfills/Object/entries/config.json:26]: done

File: polyfills/Object/entries/polyfill.js (hunk 1 above)
Comment (JakeChampion) [reply polyfills/Object/entries/polyfill.js]: yup, it was needed at first, not after I simplified it down :-)

File: polyfills/Object/entries/config.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{
+	"aliases": [
+		"es6"
```
Comment (ruiaraujo) [inline polyfills/Object/entries/config.json]: Shouldn't this be es7?

Comment (JakeChampion) [reply polyfills/Object/entries/config.json]: Good work spotting this, caught me copy pasting 🙈.
It should be `es2017`.

**Comment by JakeChampion** [comment]
Fails on CI:
```
Failures:
 • chrome/48 (targeted) [https://saucelabs.com/tests/da5d8fcd7241499dada3db7ca719bc32]
    -> Object.entries
       http://127.0.0.1:3000/test/tests?mode=targeted&feature=Object.entries
 • firefox/44 (targeted) [https://saucelabs.com/tests/24f4ef683e794f9081cbbef3d8d25327]
    -> Object.entries
       http://127.0.0.1:3000/test/tests?mode=targeted&feature=Object.entries
 • ie/13 (targeted) [https://saucelabs.com/tests/80172f6fff2642f48a0681a338785cfe]
    -> Object.entries
       http://127.0.0.1:3000/test/tests?mode=targeted&feature=Object.entries
 • ie/11 (targeted) [https://saucelabs.com/tests/7560fc116d544ced89b6fa18360a7b5b]
    -> Object.entries
       http://127.0.0.1:3000/test/tests?mode=targeted&feature=Object.entries
 • ie/10 (targeted) [https://saucelabs.com/tests/32a5c77cdfbb435a97060f2a53205ec8]
    -> Object.entries
       http://127.0.0.1:3000/test/tests?mode=targeted&feature=Object.entries
 • ie/9 (targeted) [https://saucelabs.com/tests/7109a4e346974b82b7647991ae13496f]
    -> Object.entries
       http://127.0.0.1:3000/test/tests?mode=targeted&feature=Object.entries
 • safari/9 (targeted) [https://saucelabs.com/tests/43d8271548cb4852a40814994b857846]
    -> Object.entries
       http://127.0.0.1:3000/test/tests?mode=targeted&feature=Object.entries
 • safari/8 (targeted) [https://saucelabs.com/tests/f6fec8310e654629a99ecdd5fe389c89]
    -> Object.entries
       http://127.0.0.1:3000/test/tests?mode=targeted&feature=Object.entries
 • android/5.1 (targeted) [https://saucelabs.com/tests/5995ed188630468bbc55f45ea9270cbe]
    -> Object.entries
       http://127.0.0.1:3000/test/tests?mode=targeted&feature=Object.entries
 • android/4.4 (targeted) [https://saucelabs.com/tests/1d25e71cacdc41abae43520aa4250c3e]
    -> Object.entries
       http://127.0.0.1:3000/test/tests?mode=targeted&feature=Object.entries
```

**Comment by hannesj** [comment]
@JakeChampion any update on this?

**Comment by JakeChampion** [comment]
Not yet, if you would like to take it from here you can, otherwise this will probably get looked at in the new year :-)

**Comment by hannesj** [comment]
@JakeChampion it seems there was just a small miss in removing the wrapper. My branch at https://github.com/Financial-Times/polyfill-service/pull/1025 passes on the browsers listed in on https://github.com/Financial-Times/polyfill-service/pull/976#issuecomment-262417924 in BrowserStack.

**Comment by JakeChampion** [comment]
Thanks @hannesj, we are a lot closer now :-) It seems to consistently fail on android 5.1 on circleci.

**Comment by hannesj** [comment]
The android failure seems to be due to it not following the [ES2016 Spec for ToObject](http://www.ecma-international.org/ecma-262/6.0/#sec-toobject), which is used in Object.keys, even tough it supports symbols. The test is fixed by #1026. Relates to #993.

**Comment by hannesj** [comment]
@JakeChampion Is there any estimate when you are doing a new beta release, so we could get this included in our bundle?

**Comment by JakeChampion** [comment]
@hannesj I've been attempting to cut the release since Saturday but haven't been able to get Edge 14 on BrowserStack to execute correctly. I'm seeing ` UnknownError - An unknown server-side error occurred while processing the command` all the time. This is the only thing halting a beta release being cut.

**Comment by hannesj** [comment]
Ok, good to hear that it's coming up. Hopefully BrowserStack will be working soon.

**Comment by hannesj** [comment]
@JakeChampion Any update on your solution with BrowserStack/the release?

File: polyfills/Object/entries/polyfill.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+Object.entries = function entries(object) {
+	var keys = Object.keys(object);
+
+	return keys
+		.reduce(function(entries, key) {
```
Comment (caub) [inline polyfills/Object/entries/polyfill.js]: why not .map?

**Pull request merged**